### PR TITLE
[Frontend] Add option to select visible metrics

### PR DESCRIPTION
**Pull request by Suryansh5545**
Improves Leaderboard Reading experience by only showing the default_order_by and having option to check other labels in a dropdown for them to display, also works with the order by matrix.

https://github.com/Cloud-CV/EvalAI/assets/34577232/6758508d-e2aa-4664-afbe-0f122500dbff

### CODE REVIEW

File: frontend/src/js/controllers/challengeCtrl.js [hunk 1]
```diff
@@ -1044,6 +1045,13 @@
                             vm.showSubmissionMetaAttributesOnLeaderboard = true;
                         }
 
+                        for (var k = 0; k < vm.leaderboard[i].leaderboard__schema.labels.length; k++) {
```
Comment (rahul2240) [inline frontend/src/js/controllers/challengeCtrl.js:1048]: Might wanna use this

var leaderboardLabels = vm.leaderboard[i].leaderboard__schema.labels;
var index = leaderboardLabels.findIndex(label => label === vm.orderLeaderboardBy);
var choosenmatrix = index !== -1 ? [index.toString()] : ['0'];

File: frontend/src/js/controllers/challengeCtrl.js [hunk 2]
```diff
@@ -79,6 +79,7 @@
         vm.phaseLeaderboardPublic = [];
         vm.currentPhaseLeaderboardPublic = false;
         vm.eligible_to_submit = false;
+        vm.choosenmatrix = ['0'];
```
Comment (gchhablani) [inline frontend/src/js/controllers/challengeCtrl.js:82]: `chosenmetrics`

File: frontend/tests/controllers-test/challengeCtrl.test.js [hunk 3]
```diff
@@ -853,7 +853,12 @@ describe('Unit tests for challenge controller', function () {
                     duration: 'year',
                     results: [
                         {
-                            id: 1,
```
Comment (gchhablani) [inline frontend/tests/controllers-test/challengeCtrl.test.js:856]: Did we need to remove this line?

**Review by gchhablani — COMMENTED** [review]
Added some comments.

File: frontend/tests/controllers-test/challengeCtrl.test.js (hunk 3 above)
Comment (Suryansh5545) [reply frontend/tests/controllers-test/challengeCtrl.test.js:856]: no

**Comment by gchhablani** [comment]
Can you add something like: "Visible Metrics" before the dropdown so people know what it is?

**Comment by Suryansh5545** [comment]
image

i think we should make it bold? @gchhablani ?

**Comment by Suryansh5545** [comment]
Added the label to show instead of selected values
https://github.com/Cloud-CV/EvalAI/assets/34577232/f4d993eb-7ead-424d-81f4-e3e75ac09a9b

File: frontend/src/views/web/challenge/leaderboard.html [hunk 4]
```diff
@@ -74,6 +74,12 @@ <h5 class="w-300">Leaderboard</h5>
                             <label for="getAllEntries"></label>
                             <span  class="fs-16 w-300 complete-leaderboard"> {{ challenge.getAllEntriesTestOption }}</span>
                         </div>
+                        <md-select name="leaderboarddropdownmenu" class="right" ng-model="challenge.chosenmetrics" placeholder="Visible Matrics" class="rm-margin" md-on-open="challenge.openleaderboarddropdown()" md-on-close="challenge.openleaderboarddropdown()" multiple>
+                            <md-option value="{{i}}" ng-repeat="(i, key) in challenge.leaderboard[0].leaderboard__schema.labels">
+                                <span ng-if="!challenge.leaderboarddropdown" class="fs-16 w-300">Visible Matrics</span>
```
Comment (gchhablani) [inline frontend/src/views/web/challenge/leaderboard.html:79]: `Visible Metrics` everywhere.

File: frontend/src/views/web/challenge/leaderboard.html [hunk 5]
```diff
@@ -74,6 +74,12 @@ <h5 class="w-300">Leaderboard</h5>
                             <label for="getAllEntries"></label>
                             <span  class="fs-16 w-300 complete-leaderboard"> {{ challenge.getAllEntriesTestOption }}</span>
                         </div>
+                        <md-select name="leaderboarddropdownmenu" class="right" ng-model="challenge.chosenmetrics" placeholder="Visible Matrics" class="rm-margin" md-on-open="challenge.openleaderboarddropdown()" md-on-close="challenge.openleaderboarddropdown()" multiple>
```
Comment (gchhablani) [inline frontend/src/views/web/challenge/leaderboard.html:77]: Visible Metrics

**Comment by Suryansh5545** [comment]
Video for the new look of leaderboard css
https://github.com/Cloud-CV/EvalAI/assets/34577232/3294cf44-104f-4880-afcf-b59c0bb41d5e

**Pull request merged**